algorithms/a3c/model.py

```diff
@@ -111,3 +132,94 @@ def forward(self, inputs):
         x = hx
 
         return self.critic_linear(x), self.actor_linear(x), (hx, cx)
+
+
+class A3C_LSTM_GA(torch.nn.Module):
+    """
+    Very similar to the above ActorCritic but has Gated Attention (GA) and processes an instruction
```

Comment:
I know you love vertical space, but I would split this at least into two paragraphs to help the reader. You are also saying "grounding" but not giving a clear explanation of why this is grounding or what it is.

Reply:
replaced with:

"Very similar to the above ActorCritic but has Gated Attention (GA) and processes an instruction
    which is a part of the input state using a GRU. There is also a time embedding layer to help 
    stabilize value prediction and only 3 conv layers compared to ActorCritic's 4 layers.
    Originally ran on the ViZDoom environment. The above ASCII art figures and commented code 
    below should make the flow quite clear.    
    
    The gated attention multi-modal fusion module enables the policy to focus on certain parts
    of the input image given the instruction e.g. for the instruction "Go to the red cup", a 
    specific attention filter could be learned which would enable the agent to language ground 
    itself in the meaning of both "red", "cup" and "go to". Language grounding is the ability to 
    map meaning within symbols+language into real world objects and goals. "